parsers/mcmaster-parser-2.js

```diff
@@ -18,6 +18,68 @@ function convertTo24Hour(time) {
     return time.replace(/(AM|PM)/, '');
 }
 
+
+/**
+ * Takes any date and turns it into the correct format, so javascript gets the right date.
+ * // MM/DD/YYYY is the format the javascript object reads or YYYY/MM/DD
+ * @param {String} startDate,{String} endDate
+ * @return {String Array} [startDate,endDate]
+ */
+
+function fixDateFormat(sem_start,sem_end){
+
+    if (sem_start.indexOf('-') > -1) {
+        startDate = sem_start;
+        endDate = sem_end;
+        return [startDate,endDate]
+    }
+    //assume format is either YYYY/MM/DD TO START OR MM/DD/YYYY
+    startDateSplit = sem_start.split("/");
+    endDateSplit = sem_end.split("/");
+
+    startMonth = parseInt(startDateSplit[0]);
+    startDay = parseInt(startDateSplit[1]);
+    startYear = parseInt(startDateSplit[2]);
+
+    endMonth = parseInt(endDateSplit[0]);
+    endDay = parseInt(endDateSplit[1]);
+    endYear = parseInt(endDateSplit[2]);
+
+    // This means the format is YYYY/MM/DD, because the 0th index is a year
+    if(startDateSplit[0].length == 4 && endDateSplit[0].length == 4){
+        startDate = sem_start;
+        endDate = sem_end;
+        return [startDate,endDate]
+
+    } else if (startMonth > 12 || endMonth > 12 ||
+        (startYear === endYear && startMonth > endMonth)){
+        // It must be DD/MM/YY, because month is never >12, so switch the day and month index
+        // So we need to change to format to MM/DD/YYYY so it will be interpreted well
+        startDate = [startMonth,startDay,startYear].join("/");
+        endDate = [endMonth,endDay,endYear].join("/");
+        return [startDate,endDate];
+
+        //If the different is 3 months and the semester starts in [1,5,7,9], Check is month is Jan,May,July or Sept
+    }else if ((endMonth - startMonth === 3 || endMonth - startMonth === 1) && [1,5,7,9].indexOf(startMonth) > -1) {
+        //MM/DD/YYYY it is already in the right format to nothing
+        startDate = sem_start;
+        endDate = sem_end;
+        return [startDate,endDate];
+
+        //If the different is 3 months(in this case the day is in the month slot) and the semester starts in [1,5,7,9], Check is month is Jan,May,July or Sept
+    }else if ((endMonth - startMonth === 3 || endMonth - startMonth === 1) && [1,5,7,9].indexOf(startDay) > -1) {
+        // DD/MM/YYYY is the format found because the DD index [1] slot is a Month not a day, So we need to change to format to MM/DD/YYYY so it will be interpreted well
+        startDate = [startDay,startMonth,startYear].join("/");
+        endDate = [endDay,endMonth,endYear].join("/");
+        return [startDate,endDate]
+    }else{
```

Comment:
need space between closing brace and else/else if statements, do this everywhere else as well